worker-sys/src/types/r2/uploaded_part.rs

```diff
@@ -1,14 +1,42 @@
-use wasm_bindgen::prelude::*;
+use js_sys::{Object, Reflect};
+use wasm_bindgen::JsValue;
 
-#[wasm_bindgen]
-extern "C" {
-    #[wasm_bindgen(extends=js_sys::Object)]
-    #[derive(Debug, Clone, PartialEq, Eq)]
-    pub type R2UploadedPart;
+#[derive(Debug, Clone)]
```

Comment:
Is it possible to do something like [this](https://rustwasm.github.io/wasm-bindgen/reference/types/exported-rust-types.html) here? 

Or alternatively, could you just add a constructor to the old implementation? https://rustwasm.github.io/wasm-bindgen/reference/attributes/on-js-imports/constructor.html

Reply:
Constructor was what I tried first, that didn't work because it for some reason unbeknownst to Mr wouldn't understand and the fields when passed to `complete`.

I'll give the link a tie